### ISSUE TITLE
`unformat` gives a runtime error with purescript-backend-optimizer

### ISSUE THREAD

**Issue by dariooddenino** [purescript-contrib/purescript-formatters#84]
Using `unformat` results in a runtime error

```purescript
main = do
  let res = unformat (fromFoldable [YearFull]) "2022"
  logShow res
```

`spago build && purs-backend-es bundle-app --no-build --to out.js`

This results in this runtime error: 
```
Uncaught TypeError: lift1(...) is not a function ...
...
  var unformatCommandParser = (v) => {
    if (v.tag === "YearFull") {
      return (state1, more, lift1, $$throw, done) => more((v1) => parseSignedInt(monadStateT2)(4)(exactLength)("Incorrect full year")(state1, more, lift1, $$throw, (state2, a) => more((v2) => lift1(monadStateStateT2.state((s) => $Tuple(unit, { year: $Maybe("Just", a), day: s.day, hour: s.hour, meridiem: s.meridiem, millisecond: s.millisecond, minute: s.minute, month: s.month, second: s.second })))(state2, more, lift1, $$throw, done))));
    }
...
```

I've tried creating various mimimal cases involving `mapParserT`, `State`, etc. but I was never able to reproduce this issue, sorry!

**Comment by garyb** [comment]
Thanks for trying to figure it out! :+1:

I can't look into this immediately, but will definitely do so when I can.

**Comment by garyb** [comment]
I've pretty much figured it out, it is a bug in `purescript-backend-optimizer` rather than something dodgy this library is doing. Will close this and open an issue there once I've got it down to a minimal reproduction, as it's quite tangled!

**Comment by garyb** [comment]
https://github.com/aristanetworks/purescript-backend-optimizer/issues/26

**Comment by garyb** [comment]
This was fixed in https://github.com/aristanetworks/purescript-backend-optimizer/pull/27, and was released as v1.1.1 🎉
